.github/workflows/build-wheels.yml

```diff
@@ -48,3 +48,21 @@ jobs:
       - name: Upload wheels to GH
         uses: actions/upload-artifact@v4
         path: wheelhouse/*.whl
+
+  pypi-publish:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/Wasserstein
+    permissions:
+      id-token: write
+    steps:
+      - name: Download wheels from GH
+        uses: actions/download-artifact@v4
+        with:
+          name: Packages
+          path: wheelhouse
+
+      - name: Upload package distributions to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
```

Comment:
We'll need someone who has owner control over the PypI page (@rikab) to enable trusted publishers for this to work. Instructions here: https://docs.pypi.org/trusted-publishers/adding-a-publisher/